dali/operators/decoder/audio/generic_decoder.cc

```diff
@@ -197,14 +218,19 @@ struct GenericAudioDecoder<SampleType>::Impl {
     if (!sound_) {
       throw DALIException(make_string("Failed to open encoded data: ", sf_strerror(sound_)));
     }
-
-    ret.length = sf_info_.frames;
-    ret.channels = sf_info_.channels;
-    ret.sample_rate = sf_info_.samplerate;
-    ret.channels_interleaved = true;
-    return ret;
+    return GetAudioMetadata(sf_info_);
   }
 
+  AudioMetadata OpenFromFileImpl(const std::string &filepath) {
+    DALI_ENFORCE(!filepath.empty(), "filepath is empty");
+    sf_info_ = {};
+    sound_ = sf_open(filepath.c_str(), SFM_READ, &sf_info_);
+    if (!sound_) {
+      throw DALIException(make_string("Failed to open encoded data: ", sf_strerror(sound_),
+                                      ", filepath: ", filepath));
+    }
```

Comment:
How about
```c++
DALI_ENFORCE(sound_, "...");
```

Reply:
I am just following the convention in this file (see line 198)